baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/JooqBatchWithoutBindArgs.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.BugPattern.StandardTags;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.suppliers.Supplier;
+import com.google.errorprone.suppliers.Suppliers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Type;
+import java.util.Collection;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = SeverityLevel.ERROR,
+        tags = StandardTags.PERFORMANCE,
+        summary = "Disallow jOOQ batch methods that execute queries without bind args.")
+public final class JooqBatchWithoutBindArgs extends BugChecker implements MethodInvocationTreeMatcher {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String DSL_CONTEXT = "org.jooq.DSLContext";
+    private static final String BATCH = "batch";
+    private static final String ERROR_MESSAGE =
+            "Jooq batch methods that execute queries without bind args cause performance problems.";
```

Comment:
We don't have public benchmarking data, but we do know that the way this executes is using extra CPU on the database because each query is sent over as a unique string (with all variable values inline as a string in the query). There is CPU overhead parsing and query planning each query in the batch.

Looking at how we we have used jooq internally, it was actually surprising to most of our dev teams how these calls were being executed. People usually use the batch methods to execute the same query hundreds to thousands of times but just with different bind variables. What we want to happen (and what most of our devs thought was actually happening) is it sends the query to the DB once and then just sends the arrays of bind args to execute the same query with each set of bind variables. Internally we've already refactored nearly all of our code to use the other batch methods that execute with bind args.